tests/phpunit/tests/editor/enableContentEditorForNavigationPostType.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * @group editor
+ *
+ * @covers ::_enable_content_editor_for_navigation_post_type
+ */
+class Tests_Editor_EnableContentEditorForNavigationPostType extends WP_UnitTestCase {
+
+	const NAVIGATION_POST_TYPE     = 'wp_navigation';
+	const NON_NAVIGATION_POST_TYPE = 'wp_non_navigation';
+
+	public function tear_down() {
+		$this->enable_editor_support();
+		parent::tear_down();
+	}
+
+	/**
+	 * @ticket 56266
+	 */
+	public function test_it_enables_content_editor_for_non_navigation_type_posts_after_the_content_editor_form() {
+		$this->disable_editor_support();
+		$post = $this->create_navigation_post();
+		$this->assertFalse( $this->supports_block_editor() );
+
+		_enable_content_editor_for_navigation_post_type( $post );
+
+		$this->assertTrue( $this->supports_block_editor() );
+	}
+
+	/**
+	 * @ticket 56266
+	 */
+	public function test_it_doesnt_enable_content_editor_for_non_navigation_type_posts_after_the_content_editor_form() {
+		$this->disable_editor_support();
+		$post = $this->create_non_navigation_post();
+		$this->assertFalse( $this->supports_block_editor() );
+
+		_enable_content_editor_for_navigation_post_type( $post );
+
+		$this->assertFalse( $this->supports_block_editor() );
+	}
+
+	private function supports_block_editor() {
+		return post_type_supports( static::NAVIGATION_POST_TYPE, 'editor' );
+	}
+
+	private function create_post( $post_type ) {
+		$post            = new WP_Post( new StdClass() );
```

Comment:
```suggestion
		$post            = new WP_Post( new stdClass() );
```
1. While class names are case-insensitive, for consistency and ease of searching, we should use the class names as they are named in the PHP manual.
2. Question: Should this be a `WP_Post`, or should it use the `$this->factory()->post->create()` method? I'm not sure what the answer is here.

Reply:
https://github.com/WordPress/wordpress-develop/pull/3005/files#r928870956